pkgs/development/libraries/gmime/2.nix

```diff
@@ -24,6 +26,10 @@ stdenv.mkDerivation rec {
       --replace /bin/mkdir mkdir
   '';
 
+  preConfigure = lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
+    cp ${if stdenv.hostPlatform.isMusl then ./musl-iconv-detect.h else ./iconv-detect.h} ./iconv-detect.h
```

Comment:
alternative to hardcoding `iconv-detect.h` you can invoke the emulator:
```
 "$CC" ./iconv-detect.c -o iconv-
 ${stdenv.hostPlatform.emulator buildPackages} ./iconv-detect
 rm iconv-detect
 ```
 
 i think this would save you from adding platform-specific (musl-specific) logic.

Reply:
there's no special emulator for musl so it won't work

Reply:
Can't we make it so that if `stdenv.buildPlatform.canExecute stdenv.hostPlatform`, the dummy emulator (shell) is called.

Reply:
it already does that